### PR TITLE
#245 - Add ability to specify delimiter for slice flags

### PR DESCRIPTION
**Pull request by pazderak**
Implements https://github.com/jessevdk/go-flags/issues/245

### CODE REVIEW

**Comment by pazderak** [comment]
Hi, any update on this?

**Comment by martynasb** [comment]
- @jessevdk gentle ping
- @pazderak This PR title should reflect you are also changing the delimiter for maps

**Comment by pazderak** [comment]
@jessevdk gentle reminder to merge this PR?

File: convert.go [hunk 1]
```diff
@@ -136,7 +136,11 @@ func convertToString(val reflect.Value, options multiTag) (string, error) {
 				return "", err
 			}
 
-			ret += keyitem + ":" + item
+			delim := options.Get("value-delim")
```
Comment (jessevdk) [inline convert.go]: I don't think this should affect maps. I want consistency over configurability

File: convert.go [hunk 2]
```diff
@@ -249,16 +253,28 @@ func convert(val string, retval reflect.Value, options multiTag) error {
...
+
+			if err := convert(p, elemval, options); err != nil {
+				return err
+			}
+
+			retval.Set(reflect.Append(retval, elemval))
+		}
 	case reflect.Map:
-		parts := strings.SplitN(val, ":", 2)
+		delim := options.Get("value-delim")
```
Comment (jessevdk) [inline convert.go]: Ditto, don't think we should do this for maps

File: convert.go [hunk 3]
```diff
@@ -249,16 +253,28 @@ func convert(val string, retval reflect.Value, options multiTag) error {
 	case reflect.Slice:
 		elemtp := tp.Elem()
 
-		elemvalptr := reflect.New(elemtp)
-		elemval := reflect.Indirect(elemvalptr)
-
-		if err := convert(val, elemval, options); err != nil {
-			return err
+		parts := []string{val}
+		delim := options.Get("value-delim")
```
Comment (jessevdk) [inline convert.go]: We need to think about the case of users wanting to specify also a value that contains the delim (i.e. escaping the delim).

**Comment by oguzyildiz1991** [comment]
@pazderak any updates on this? I could really use it in my case :)

**Comment by dolmen** [comment]
`separator` would be a better name than `value-delim`...

**Pull request closed without merging**